### PR TITLE
Registry

### PR DESCRIPTION
**Pull request by raphink**
This rebases and finalizes the registry feature.

I'm sorry I just copied yours files, fixed them and re-commited them to avoid complex rebasing.

### CODE REVIEW

File: rancher/resource_rancher_registry_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+const testAccRancherRegistryConfig = //`
+// resource "rancher_environment" "foo_registry" {
+//   name = "registry test"
+//   description = "environment to test registries"
+// }
+//
+// resource "rancher_environment" "foo_registry2" {
+//   name = "alternative registry test"
+//   description = "other environment to test registries"
+// }
```
Comment (blackjid) [inline rancher/resource_rancher_registry_test.go]: what do you thing if we remove this now we are using the default environment??

Comment (blackjid) [reply rancher/resource_rancher_registry_test.go]: you can squash that into the first commit

Comment (raphink) [reply rancher/resource_rancher_registry_test.go]: OK, sure.

Comment (raphink) [reply rancher/resource_rancher_registry_test.go]: Done.

**Pull request merged**